### PR TITLE
Adding sync and catalog endpoints

### PR DESCRIPTION
**Pull request by OrenVildermanPepperi**

### CODE REVIEW

File: entities/sync.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export interface SyncJobInfo {
+    SyncUUID: string;
+    CreationDateTime: number;
+    ModificationDateTime: number;
+    CreatedByUserUUID: string;
+    Status: string;
+    ProgressPercentage: number;
+    SentData: {
+        ResponseURL?: string;
+        DataFullyUpdated?: boolean;
```
Comment (Chasky-H) [inline entities/sync.ts:33]: Why are these optional

File: entities/sync.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    } | null;
+    DataUpdates: {
+        URL?: string;
```
Comment (Chasky-H) [inline entities/sync.ts:36]: Why is this optional

File: entities/sync.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    } | null;
+    FreeTextFromClient?: string;
```
Comment (Chasky-H) [inline entities/sync.ts:38]: Shouldn't this be in the SyncBody as well?

File: entities/sync.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    TimeZoneDiff?: number;
+    Locale?: true;
+    BrandedAppID?: number;
+    UserFullName?: string;
+    SoftwareVersion?: number;
+    SourceType?: string;
+    DeviceModel?: string;
+    DeviceName?: string;
+    DeviceScreenSize?: number;
+    SystemName?: string;
```
Comment (Chasky-H) [inline entities/sync.ts:15]: Are you sure that all these are optional?

File: entities/sync.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+export interface SyncBody {
+    LocalDataUpdates: any | null;
```
Comment (Chasky-H) [inline entities/sync.ts:2]: Please specify the correct scheme of this parameter

Comment (OrenVildermanPepperi) [reply entities/sync.ts]: After the Talk we had to day, I fixed SyncBody in a new pull request

**Comment by OrenVildermanPepperi** [comment]
bumped minor version

**Pull request merged**